sbin/start-history-server.sh

```diff
@@ -30,10 +30,11 @@ sbin="`cd "$sbin"; pwd`"
 . "$sbin/spark-config.sh"
 . "$SPARK_PREFIX/bin/load-spark-env.sh"
 
-if [ $# != 0 ]; then
+if [ $# != 0 ] && [[ ! "$1" == \-* ]]; then
```

Comment:
I think this works, though I also wonder if it's possible to parse this in `HistoryServerArguments`. Like, check the first non-flag argument after all flags have been parsed. Override the dir if present, and emit a warning. I haven't looked into it, maybe it's not easy, but worth ruling out.